test/integration_test.go

```diff
@@ -72,6 +73,65 @@ func TestCreateResource(t *testing.T) {
 		Done()
 }
 
+func TestUpdateResource(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skpping integration test in short mode")
+	}
+
+	byt, err := ioutil.ReadFile("../examples/request.json")
+	if err != nil {
+		panic(err)
+	}
+	var postData map[string]interface{}
+
+	if err := json.Unmarshal(byt, &postData); err != nil {
+		panic(err)
+	}
+
+	setupTest().Post("/v1/resource").
+		SetHeader("Foo", "Bar").
+		JSON(postData).
+		Expect(t).
+		Status(201).
+		Type("json").
+		JSONSchema(resourceSchema).
+		AssertFunc(assertResourceResponse).
+		Done()
+
+	byt, err = ioutil.ReadFile("../examples/update_request.json")
+	if err != nil {
+		panic(err)
+	}
+
+	var patchData map[string]interface{}
+
+	if err := json.Unmarshal(byt, &patchData); err != nil {
+		panic(err)
+	}
+
+	setupTest().Patch(fmt.Sprintf("/v1/resource/%s", id)).
+		SetHeader("Content-Type", "application/json").
+		JSON(patchData).
+		Expect(t).
+		Status(200).
+		Type("json").
+		JSONSchema(resourceSchema).
+		AssertFunc(assertResourceResponse).
+		Done()
+
+	setupTest().Get(fmt.Sprintf("/v1/resource/%s", id)).
+		Expect(t).
+		Status(200).
+		Type("json").
+		AssertFunc(assertUpdatedResourceResponse).
+		Done()
+
+		// minimal integration test for label (the only field changed in our update statement)
+	if label != patchData["label"] {
```

Comment:
can we move this assertion into `assertUpdatedResourceResponse`?  Othewise we're doing a lot of gymnastics for `label` to be available in this context.

Reply:
That makes sense. Based on the comment of the other method, I was under the impression this was being used to set variable, so misused it here.

Reply:
On second pass, I'm not sure this is any less convoluted work. I don't think (initially) `assertFunc` is setup this way. Will work through, but it's something we may want to walk through together tomorrow.